src/80850_len_3060.c

```diff
@@ -454,76 +450,611 @@ void status_menu_draw_number(s32 iconID, s32 x, s32 y, s32 value, s32 numDigits)
+                uiStatus->displaySP = playerData->specialBarsFilled;
+            }
+        }
+    }
+
+    if (uiStatus->disabled) {
+        return;
+    }
+
+    if (uiStatus->unk_45[1] && uiStatus->hidden && playerStatus->inputEnabledCounter == 0) {
+        uiStatus->showTimer = 42;
+        uiStatus->hidden = FALSE;
+        uiStatus->unk_3B[0] = 0;
+    }
+
+    switch (uiStatus->hidden) {
+        case 0:
+            uiStatus->drawPosY += 10;
+            if (uiStatus->drawPosY >= 18) {
+                uiStatus->drawPosY = 18;
+                if (uiStatus->unk_3B[1] && uiStatus->unk_3B[0] && playerStatus->actionState != ACTION_STATE_IDLE) {
+                    uiStatus->showTimer = 0;
+                }
+                if (uiStatus->showTimer != 0) {
+                    uiStatus->showTimer--;
+                } else {
+                    if (!uiStatus->ignoreChanges) {
+                        if (!uiStatus->unk_3B[0] || playerStatus->actionState != ACTION_STATE_IDLE) {
+                            if (gGameStatusPtr->isBattle == 0) {
+                                uiStatus->hidden = 1;
+                                uiStatus->showTimer = 0;
+                                uiStatus->unk_3B[1] = 0;
+                            }
+                        }
+                    }
+                }
+            }
+            break;
+        case 1:
+            uiStatus->drawPosY -= 5;
+            if (uiStatus->drawPosY < -100) {
+                uiStatus->drawPosY = -100;
+                if (!uiStatus->ignoreChanges) {
+                    if (playerStatus->actionState != ACTION_STATE_IDLE) {
+                        uiStatus->showTimer = 0;
+                    } else if (playerStatus->flags & (PS_FLAGS_NO_STATIC_COLLISION | PS_FLAGS_INPUT_DISABLED)) {
+                        uiStatus->showTimer = 0;
+                    } else {
+                        uiStatus->showTimer++;
+                    }
+
+                    if (uiStatus->showTimer >= 240 && gGameStatusPtr->isBattle == 0) {
+                        uiStatus->showTimer = 210;
+                        uiStatus->hidden = 0;
+                        uiStatus->unk_3B[0] = 1;
+                        uiStatus->unk_3B[1] = 1;
+                    }
+                }
+            }
+            break;
+    }
+
+    gDPSetScissor(gMasterGfxPos++, G_SC_NON_INTERLACE, 12, 20, SCREEN_WIDTH - 12, SCREEN_HEIGHT - 20);
+    fp = uiStatus->drawPosX;
+    s5 = uiStatus->drawPosY;
+    draw_box(0, 5, fp,       s5, 0, 174, 35, 255, 0, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, NULL, NULL, NULL, 320, 240, NULL);
+    draw_box(0, 6, fp + 174, s5, 0, 122, 25, 255, 0, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, NULL, NULL, NULL, 320, 240, NULL);
+
+    if (uiStatus->hpBlinkTimer > 0) {
+        uiStatus->hpBlinkTimer--;
+        if (uiStatus->hpBlinkTimer == 0) {
+            status_menu_stop_blinking_hp();
+        }
+    }
+
+    s6 = 1;
```

Comment:
Is s6 a boolean? It's only ever 1 or 0 and line 716 and others use it conditionally.

Reply:
This looks interesting, might be better to do a documentation pass later when we're more sure